scanspec/regions.py

```diff
@@ -259,26 +317,96 @@ class Circle(Region):
         metadata=schema(description="The name matching the x axis of the spec")
     )
     y_axis: str = field(
-        metadata=schema(description="The name matching the x axis of the spec")
+        metadata=schema(description="The name matching the y axis of the spec")
     )
-    x_centre: float = field(
-        metadata=schema(description="Minimum inclusive x value in the region")
+    x_middle: float = field(
+        metadata=schema(description="The central x point of the circle")
     )
-    y_centre: float = field(
-        metadata=schema(description="Minimum inclusive y value in the region")
+    y_middle: float = field(
+        metadata=schema(description="The central y point of the circle")
+    )
+    radius: float = field(
+        metadata=schema(description="Radius of the circle", exc_min=0)
     )
-    radius: float = field(metadata=schema(description="Radius of the circle"))
 
     def axis_sets(self) -> List[Set[str]]:
         return [{self.x_axis, self.y_axis}]
 
     def mask(self, points: AxesPoints) -> np.ndarray:
-        x = points[self.x_axis] - self.x_centre
-        y = points[self.y_axis] - self.y_centre
+        x = points[self.x_axis] - self.x_middle
+        y = points[self.y_axis] - self.y_middle
         mask = x * x + y * y <= (self.radius * self.radius)
         return mask
 
 
+@dataclass
+class Ellipse(Region):
+    """Mask contains points of axis within an xy ellipse of given radius
+
+    .. example_spec::
+
+        from scanspec.plot import plot_spec
+        from scanspec.specs import Line
+        from scanspec.regions import Ellipse
+
+        #Ellipse parameters
+        x_axis = "x"
+        y_axis = "y"
+        x_middle = 5
+        y_middle = 5
+        x_semiaxis = 2
+        y_semiaxis = 3
+        angle = 75
+
+        grid = Line(y_axis, 3, 8, 10) * ~Line(x_axis, 1 ,8, 10)
+        spec = grid & Ellipse(x_axis, y_axis, x_middle, y_middle, x_semiaxis,
+        y_semiaxis, angle)
+        plot_spec(spec)
+    """
+
+    x_axis: str = field(
+        metadata=schema(description="The name matching the x axis of the spec")
+    )
+    y_axis: str = field(
+        metadata=schema(description="The name matching the y axis of the spec")
+    )
+    x_middle: float = field(
+        metadata=schema(description="The central x point of the ellipse")
+    )
+    y_middle: float = field(
+        metadata=schema(description="The central y point of the ellipse")
+    )
+    x_semiaxis: float = field(
```

Comment:
Hmm, the naming doesn't seem right, I've never heard of a semiaxis before, but we can't use width and height here as we are generic in x and y. How about we specify x_min, y_min, x_max, y_max like for a rectangle rather than the midpoints like the circle?

Reply:
Alternatively, x_semidiameter and y_semidiameter might be appropriate technical terms https://en.wikipedia.org/wiki/Semidiameter.

Reply:
https://en.wikipedia.org/wiki/Semi-major_and_semi-minor_axes

Reply:
Though realistically using radius is the most inclusive term

Reply:
x_radius and y_radius?